provider/cmd/helpers.go

```diff
@@ -58,3 +71,49 @@ func providerFromFlags(flags *pflag.FlagSet) (sdk.Address, error) {
 
 	return addr, nil
 }
+
+func providersFromFlagsOrQuery(flags *pflag.FlagSet, ctx context.Context, cctx client.Context, did dtypes.DeploymentID) ([]sdk.Address, error) {
+	var providers []string
+
+	if !flags.Changed(FlagProvider) {
+		cclient := akashclient.NewQueryClientFromCtx(cctx)
+		resp, err := cclient.Leases(ctx, &mtypes.QueryLeasesRequest{
```

Comment:
It should filter leases by whatever you pass.  An owner and dseq is required, but oseq, gseq, provider can be used to specify it further.

It shouldn't accept a random provider like is done below.